util-cgroups/tests/integration.rs

```diff
@@ -0,0 +1,892 @@
+
+    let cpt: std::sync::Arc<AtomicU8> = std::sync::Arc::new(AtomicU8::new(0));
+    let ac_out1 = std::sync::Arc::clone(&cpt);
+    let ac_out2 = std::sync::Arc::clone(&cpt);
+
+    // Detection of Cgroup
+    let config1 = Config::default();
+    let config2 = Config::default();
+    let f1 = callback(move |cgroups| {
+        assert_eq!(cgroups.len(), 1);
+        for cgroup in cgroups {
+            assert!(cgroup.fs_path().iter().any(|part| part.to_os_string() == "cpuacct"));
+            ac_out1.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        }
+        Ok(())
+    });
+    let f2 = callback(|_a| Ok(()));
+    let handler = ClosureCallbacks {
+        on_cgroups_created: f1,
+        on_cgroups_removed: f2,
+    };
+    let detector = CgroupDetector::new(hierarchy_cpu, config1, handler);
+    assert!(detector.is_ok());
+
+    let f3 = callback(move |cgroups| {
+        assert_eq!(cgroups.len(), 1);
+        for cgroup in cgroups {
+            assert!(cgroup.fs_path().iter().any(|part| part.to_os_string() == "memory"));
+            ac_out2.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        }
+        Ok(())
+    });
+    let handler2 = ClosureCallbacks {
+        on_cgroups_created: f3,
+        on_cgroups_removed: f2,
+    };
+    let detector = CgroupDetector::new(hierarchy_mem, config2, handler2);
+    assert!(detector.is_ok());
+
+    assert_eq!(2, cpt.load(std::sync::atomic::Ordering::SeqCst));
+}
+
+#[test]
+fn one_cgroup_created_after_v1() {
+    env_logger::init_from_env(env_logger::Env::default());
+    let root_unwraped = tempdir().unwrap();
+    let root = root_unwraped.path().to_path_buf();
+    let version = CgroupVersion::V1;
+    // Create path before
+    let path_cpu = root.join("cpuacct");
+    //Creation of CgroupHierarchy
+    let hierarchy_cpu = CgroupHierarchy::manually_unchecked(path_cpu, version, vec!["cpuacct"]);
+    let res_cpu = create_folder(&root, "cpuacct", hierarchy_cpu.clone()).unwrap();
+    let cpt: std::sync::Arc<AtomicU8> = std::sync::Arc::new(AtomicU8::new(0));
+    let ac_out = std::sync::Arc::clone(&cpt);
+    // Detection of Cgroup
+    let config = Config {
+        v1_refresh_interval: Duration::from_millis(10),
+        force_polling: false,
+    };
+    let f1 = callback(move |cgroups| {
+        for cgroup in cgroups {
+            assert!(cgroup.fs_path().iter().any(|part| part.to_os_string() == "cpuacct"));
+            ac_out.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        }
+        Ok(())
+    });
+    let f2 = callback(|_a| Ok(()));
+    let handler = ClosureCallbacks {
+        on_cgroups_created: f1,
+        on_cgroups_removed: f2,
+    };
+    let detector = CgroupDetector::new(hierarchy_cpu.clone(), config, handler);
+    assert!(detector.is_ok());
+    // Creation of cgroup
+    let _res_cpu = create_folder(&res_cpu, "toto", hierarchy_cpu.clone());
+    thread::sleep(Duration::from_secs(5));
```

Comment:
It would be great if this duration was a `const`ant, so we can change it for every test at once.